bokeh/models/actions.py

```diff
@@ -15,3 +15,25 @@ class OpenURL(Action):
     The URL to direct the web browser to. This can be a template string,
     which will be formatted with data from the data source.
     """)
+
+class Callback(Action):
+    """ Execute a JavaScript function. """
+
+    args = Dict(String, Instance(PlotObject), help="""
+    A mapping of names to Bokeh plot obejcts. These objects are made
+    available to the callback code snippet as the values of named
+    parameters to the callback.
+    """)
+
+    code = String(help="""
+    A snippet of JavaScript code to execute in the browser. The code is
+    made into the body of a function, and all of of the named objects in
+    ``args`` are available as parameters that the code can use. Additionally,
+    a ``cb_obj`` parameter contains the object that triggered the callback.
+
+    .. warning:
+        Due to JS context and scoping rules, if the code snippet calls
+        external functions that use non-"var" variables with names that
+        conflict with names in ``args``, there is the possibility of subtle
+        errors.
```

Comment:
Ok, here's the adjusted code that is the case I was worried about.  It takes two bad implementations (both functions have to access the global scope).  If you change `func2` to take `source` as an argument it's bound to the scope and `func1` is executed with `source` being undefined.

```javascript
var func1 = function() {
  console.log("func1: " + source);
  source = "set by func1";
  console.log("func1: " + source);
}

var func2 = function() {
  source = "set by func2";
  console.log("func2: " + source);
  func1();
  console.log("func2: " + source);
}

func2();
```

Next time I'll write a test case instead of a *text* case. :-)